python/thunder/rdds/series.py

```diff
@@ -56,6 +56,22 @@ def index(self):
         if self._index is None:
             self.populateParamsFromFirstRecord()
         return self._index
+        
+    @index.setter
+    def index(self, value):
+        try:
+            value[0]
+        except:
+            raise ValueError("Index must be a non-empty indexable object")
+        if self._index is None:
+            self.populateParametersFromFirstRecord()
+        try:
+            lenMatch = len(value) == len(self._index)
+        except:
+            raise ValueError("Index must be an object with a length")
+        if not lenMatch:
+            raise ValueError("New index length must match that of the current index")
```

Comment:
Oh sorry - that of course should have been:
```python
try:
  if len(val) != len(self._index):
    raise ValueError(blah blah)
except TypeError:
  raise TypeErrorOrSomething(another more descriptive blah)
```
Or you could just let the initial TypeError propagate on through. But whatever, really not a big deal. :)